File: backend/GameApp.Application/UseCases/GameUseCases/GameUpdateUseCase.cs
using GameApp.Domain.Entities;
using GameApp.Domain.Entities.Scenes;
using GameApp.Domain.Enumerates;
using GameApp.Domain.ValueObjects.Characters;
using GameApp.Domain.ValueObjects.Enemies;

namespace GameApp.Application.UseCases.GameUseCases;

public interface GameUpdateUseCase
{
    public Task<Game?> UpdateGame(Guid id, Character character, int numberScenesToFinish, List<Scene> completedScenes, NothingHappensScene finalScene, List<Scene> listCurrentScenes, List<UserAction> listCurrentUserActions, GameStatus status, Enemy? currentEnemy);
}

File: frontend/gameapp-frontend/src/application/usecases/game-use-cases/game-update-use-case.ts
import { Character } from '../../../domain/value-objects/characters/character';
import { Scene } from '../../../domain/entities/scenes/scene';
import { Enemy } from '../../../domain/entities/enemy';
import { Game } from '../../../domain/entities/game';
import type { GameStatus } from '../../../domain/enumerates/game-status';
import type { FinalScene } from '../../../domain/entities/scenes/final-scene';
import type { GameDifficulty } from '../../../domain/enumerates/game-difficulty';

export interface IGameUpdateUseCase {
    updateGame(
        id: string,
        difficulty: GameDifficulty,
        character: Character,
        numberScenesToFinish: number,
        finalScene: FinalScene,
        listCurrentScenes: Scene[],
        listCurrentUserActions: string[],
        completedScenes: Scene[],
        status: GameStatus,
        currentEnemy: Enemy | null
    ): Promise<Game>;
}
